core/tee/tee_svc_cryp.c

```diff
@@ -3386,13 +3436,25 @@ TEE_Result syscall_asymm_operate(unsigned long state,
 	switch (cs->algo) {
 	case TEE_ALG_RSA_NOPAD:
 		if (cs->mode == TEE_MODE_ENCRYPT) {
-			res = crypto_acipher_rsanopad_encrypt(o->attr, src_data,
+			if (!(o->info.objectUsage & TEE_USAGE_EXTRACTABLE)) {
```

Comment:
I think that extending the key object attribute with a `TEE_ATTR_PRIVATE_KEY_ENCRYPTED` or `TEE_ATTR_PRIVATE_KEY_NOT_PLAINTEXT` would let more flexibility.
- No need to add new APIs to use a private key "not plaintext". This is a key attribute directly managed by the final crypto driver. (In fact this is the final driver using the key that implement the correct logic or not if not supported).
- As this is the crypto driver that know the OVERHEAD to add, no need to change the bignumber module. (In case of the caam driver, when all asymmetric algo will be done, we will be able to remove bignumber dependency)
- Finally this could be applicable to other asymmetric algo like ECC, DSA supporting encrypted key too.

Is it a possible option?

Reply:
Hi @cneveux 

Actually we don't want to add any new object type or attributes, if added we need to get this added in Global Platform TEE Internal APIs.
Even if we extend the attributes, I didn't understood your first point, how can we avoid adding new APIs, Can you please explain this point.

Reply:
@sahilnxp 

If the rsa keypair structure get a field saying that the Private Key is encrypted or whatever you want indicating that key is not plaintext. Because crypto_acipher_rsaXXX first parameter (o->attr) is the key (either public or keypair) the API knows immediately by checking this field what to do.

Reply:
@cneveux 
I got your point.

> As this is the crypto driver that know the OVERHEAD to add, no need to change the bignumber module.

Regarding this, I got it that crypto driver knows the OVERHEAD and it will take care of it, but in the end the data generated by crypto driver needs to be copied into the memory that is being allocated in `crypto_acipher_alloc_rsa_keypair()` in case of RSA so we need to allocate more memory.

Or may be you are talking about allocating only a simple memory buffer instead of bignums for holding the private parts of RSA key ??

Reply:
Agree with you concerning the memory allocation. Could I advice you to have a look to the #3591 which is the CAAM RSA driver? You'll will certainly understand better. The final driver is doing the memory allocation function of the needs. 
Bignumber library is not a best to use module for the CAAM because we need to do big memory copy (to do conversion). This is why I said that at the end when bignumber will not more needed for the CAAM driver, we will no more use bignumber when not necessary.